### PR TITLE
Implemented | Web hooks card on Product View Page

### PR DESCRIPTION
**Pull request by rvutd**
### Related Issues
Implement webhook configuration #175

Closes #

### Short Description and Why It's Useful
Users can know if they have subscribed to a topic or not.

#### Before: 

Screenshot 2022-05-26 at 1 03 49 PM

### After:
Screenshot 2022-05-26 at 1 05 55 PM

#### Before:
Screenshot 2022-05-29 at 6 37 55 PM

#### After:
Screenshot 2022-05-29 at 6 36 54 PM

**IMPORTANT NOTICE** - Remember to add a changelog entry

### Contribution and Currently Important Rules Acceptance

- [x] I read and followed [contribution rules](https://github.com/hotwax/job-manager#contribution-guideline)

### CODE REVIEW

File: src/store/modules/webhooks/actions.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import WebhooksState from "./WebhooksState";
+import { WebhooksService } from "@/services/WebhooksService";
+import { hasError } from "@/utils";
+
+const actions: ActionTree<WebhooksState, RootState> = {
+  async fetchWebhooks({ commit }, payload){
+    await WebhooksService.fetchShopifyWebhooks(payload).then(resp => {
+      if (resp.status === 200 && resp.data.webhooks?.length > 0 && !hasError(resp)) {
+        const { webhooks } = JSON.parse(resp.data.webhooks);
+        console.log(webhooks);
```
Comment (ymaheshwari1) [inline src/store/modules/webhooks/actions.ts]: Unwanted console log statement

File: src/store/modules/webhooks/actions.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        
+        const topics: any = {}
+        webhooks.forEach((topic: any) => {
```
Comment (ymaheshwari1) [inline src/store/modules/webhooks/actions.ts]: Use map instead.

File: src/store/modules/webhooks/actions.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+          topics[topic.topic] = topic
+        })
+        commit('setWebhooks', topics)
+      } else {
+        console.log('something went wrong');
```
Comment (ymaheshwari1) [inline src/store/modules/webhooks/actions.ts:24]: Unwanted console statement.

File: src/store/modules/webhooks/getters.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { GetterTree } from "vuex";
+import RootState from "@/store/RootState";
+import WebhooksState from "./WebhooksState";
+
+const getters: GetterTree<WebhooksState, RootState> = {
+  fetchCachedWebhooks: (state) => state.cached
```
Comment (ymaheshwari1) [inline src/store/modules/webhooks/getters.ts]:
```suggestion
  getCachedWebhooks: (state) => state.cached
```

File: src/store/modules/webhooks/mutations.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import { MutationTree } from "vuex";
+import WebhooksState from "./WebhooksState";
+
+const mutations: MutationTree<WebhooksState> = {
+  setWebhooks (state, payload: any) {
```
Comment (ymaheshwari1) [inline src/store/modules/webhooks/mutations.ts:5]: Define mutation types and then use the types, take reference from job module

File: src/views/Orders.vue [hunk 6]
```diff
@@ -36,6 +36,28 @@
             </ion-item>
           </ion-card>
 
+          <ion-card>
+            <ion-card-header>
+              <ion-card-title>{{ $t("Webhooks") }}</ion-card-title>
+            </ion-card-header>
+            <ion-item>
+              <ion-label class="ion-text-wrap">{{ $t("New orders") }}</ion-label>
+              <ion-toggle slot="end" color="secondary"></ion-toggle>
```
Comment (ymaheshwari1) [inline src/views/Orders.vue]: Use self-closing tag.

File: src/views/Product.vue [hunk 7]
```diff
@@ -106,7 +108,8 @@ export default defineComponent({
       currentJobStatus: '',
       freqType: '',
       isJobDetailAnimationCompleted: false,
-      isDesktop: isPlatform('desktop')
+      isDesktop: isPlatform('desktop'),
+      webhooksEnums: JSON.parse(process.env?.VUE_APP_WEBHOOKS_ENUMS as string) as any
```
Comment (ymaheshwari1) [inline src/views/Product.vue:112]:
```suggestion
      webhookEnums: JSON.parse(process.env?.VUE_APP_WEBHOOKS_ENUMS as string) as any
```

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: .env.example [hunk 8]
```diff
@@ -11,4 +11,5 @@ VUE_APP_INITIAL_JOB_ENUMS={"IMP_PRDTS_BLK":"JOB_IMP_PROD_NEW_BLK","IMP_ORDERS_BL
...
\ No newline at end of file
+VUE_APP_BATCH_JOB_ENUMS={"JOB_BKR_ORD_UNF":{"id":"JOB_BKR_ORD_UNF","facilityId":"_NA_","unfillable": true},"JOB_BKR_ORD":{"id": "JOB_BKR_ORD","facilityId":"_NA_","unfillable": false},"JOB_BKR_PREORD_UNF":{"id":"JOB_BKR_PREORD_UNF","facilityId":"PRE_ORDER_PARKING","unfillable":true},"JOB_BKR_PREORD":{"id":"JOB_BKR_PREORD","facilityId":"PRE_ORDER_PARKING","unfillable":false},"JOB_BKR_BACKORD_UNF":{"id":"JOB_BKR_BACKORD_UNF","facilityId":"BACKORDER_PARKING","unfillable":true},"JOB_BKR_BACKORD":{"id":"JOB_BKR_BACKORD","facilityId":"BACKORDER_PARKING","unfillable":false}}
+VUE_APP_WEBHOOKS_ENUMS={"NEW_PRODUCTS":"products/create","DELETE_PRODUCTS":"products/update","NEW_ORDERS":"orders/create","CANCELLED_ORDERS":"orders/cancelled","PAYMENT_STATUS":"orders/paid","RETURNS":""}
```
Comment (ymaheshwari1) [inline .env.example:15]:
```suggestion
VUE_APP_WEBHOOK_ENUMS={"NEW_PRODUCTS":"products/create","DELETE_PRODUCTS":"products/update","NEW_ORDERS":"orders/create","CANCELLED_ORDERS":"orders/cancelled","PAYMENT_STATUS":"orders/paid","RETURNS":""}
```

File: src/store/modules/webhooks/index.ts [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+import WebhooksState from './WebhooksState'
+import RootState from '@/store/RootState'
+import getters from './getters'
+import mutations from './mutations'
+import actions from './actions'
+
+const webhooksModule: Module<WebhooksState, RootState> = {
+  namespaced: true,
+  state: {
+    cached: {
```
Comment (ymaheshwari1) [inline src/store/modules/webhooks/index.ts:11]: I think we can define the cached state as empty.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/store/modules/webhook/actions.ts [hunk 10]
```diff
@@ -0,0 +1,29 @@
+import { ActionTree } from "vuex";
+import RootState from "@/store/RootState";
+import WebhookState from "./WebhookState";
+import { WebhookService } from "@/services/WebhookService";
+import { hasError } from "@/utils";
+import * as types from './mutations-types'
+
+const actions: ActionTree<WebhookState, RootState> = {
+  async fetchWebhooks({ commit }, payload){
+    await WebhookService.fetchShopifyWebhooks(payload).then(resp => {
```
Comment (ymaheshwari1) [inline src/store/modules/webhook/actions.ts:10]: Handle the case when API returns an error like status 500

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/store/modules/webhooks/actions.ts (hunk 1 above)
Comment (rvutd) [reply src/store/modules/webhooks/actions.ts]: updated

File: src/store/modules/webhooks/actions.ts (hunk 2 above)
Comment (rvutd) [reply src/store/modules/webhooks/actions.ts]: updated

File: src/store/modules/webhooks/getters.ts (hunk 4 above)
Comment (rvutd) [reply src/store/modules/webhooks/getters.ts]: updated

File: src/store/modules/webhooks/actions.ts (hunk 3 above)
Comment (rvutd) [reply src/store/modules/webhooks/actions.ts]: updated

File: src/store/modules/webhooks/mutations.ts (hunk 5 above)
Comment (rvutd) [reply src/store/modules/webhooks/mutations.ts]: updated

File: src/views/Orders.vue (hunk 6 above)
Comment (rvutd) [reply src/views/Orders.vue]: updated

File: src/views/Product.vue (hunk 7 above)
Comment (rvutd) [reply src/views/Product.vue]: updated

File: .env.example (hunk 8 above)
Comment (rvutd) [reply .env.example]: updated

File: src/store/modules/webhook/mutations-types.ts [hunk 11]
```diff
@@ -0,0 +1,2 @@
+export const SN_WEBHOOK = 'webhook'
+export const WEBHOOK_UPDATED = SN_WEBHOOK + 'SET_WEBHOOK'
```
[the remark this answers is not in the document]
Comment (rvutd) [reply src/store/modules/webhook/mutations-types.ts]: updated

File: src/store/modules/webhooks/index.ts (hunk 9 above)
Comment (rvutd) [reply src/store/modules/webhooks/index.ts]: updated

**Pull request merged**